### ISSUE TITLE
putAll() false sharing issue

### ISSUE THREAD

**Issue by vjeko** [CorfuDB/CorfuDB#574]
```java
  @Override
  @Mutator(
      name = "putAll"
  )
  public void putAll(Map<? extends K, ? extends V> m) {
    proxy_CORFUSMR.logUpdate("putAll",false,null,m);
  }
```

**Comment by Maithem** [comment]
Ok. Do you have a test case that surfaces this bug?

**Comment by no2chem** [comment]
Can you give more detail? When is transaction safety violated? (Performance may suffer until #508 is merged, perhaps we should break that PR up).




On Mon, May 01, 2017 at 5:57pm, Vjekoslav Brajkovic < notifications@github.com [notifications@github.com] > wrote:
@Override@Mutator(
      name="putAll"
  )
  publicvoid putAll(Map<? extends K, ? extends V> m) {
    proxy_CORFUSMR.logUpdate("putAll",false,null,m);
  }

**Comment by vjeko** [comment]
Maithem, take a look at `NSXPlatformTransactionManagerIT::testMutatorOnly()`

**Comment by vjeko** [comment]
Here is a quote from my email to Michael:

> In one of my tests I've noticed that putAll() is aborting a transaction despite the fact that there are no conflicts. Upon further investigation, I have noticed that putAll is missing @ConflictParameter annotation. Are there any plans to add conflict resolution to this this API call?

**Comment by no2chem** [comment]
Yes, the PR I cited, which hasn't been merged in should fix this, I'll look into it this evening



On Mon, May 01, 2017 at 6:26pm, Vjekoslav Brajkovic < notifications@github.com [notifications@github.com] > wrote:
Here is a quote from my email to Michael:

In one of my tests I've noticed that putAll() is aborting a transaction despite the fact that there are no conflicts. Upon further investigation, I have noticed that putAll is missing @ConflictParameter annotation. Are there any plans to add conflict resolution to this this API call?
